Mathlib/Util/GetAllModules.lean

```diff
@@ -40,7 +40,11 @@ def getAllFiles (git : Bool) (ml : String) : IO (Array System.FilePath) := do
     else do
       let all ← walkDir ml
       return all.filter (·.extension == some "lean"))
-  let files := (allModules.erase ml.lean).qsort (·.toString < ·.toString)
+  -- sort after replacing the `pathSeparators` by `/`, to avoid sorting issues between `\` and `/`
+  let files := (allModules.erase ml.lean).qsort fun f1 f2 =>
+    let f1 := f1.toString.replace ("".push pathSeparator) "/"
```

Comment:
Is it sufficient to just replace the "ideal" path separator, or is there need to replace all path separators? (There is [System.FilePath.pathSeparators](https://leanprover-community.github.io/mathlib4_docs/Init/System/FilePath.html#System.FilePath.pathSeparators) also.) Perhaps the APIs called before ensure to only use the ideal one, so this code works --- in any case, this subtlety seems useful to document. Yes, I would also prefer for it to not exist in the first place.

Reply:
I am not really sure how to answer this: the code works on my computer (running Linux) and on CI (running again some Unix-based system).  I think that testing it on a Mac and on a Windows computer would be desirable, to make sure that the ordering there is the same as here.

Reply:
Agreed testing on other systems is worthwhile - perhaps ask on zulip after this PR lands?
That said: I think replacing all possible path separators (use `FilePath.pathSeparators`) by '/' is sufficient. (Optional, "the extra mile": ask on zulip of walkdir makes or wants to make a guarantee that all paths returned use the "ideal" path separator.)

Reply:
I looped over the pathSeparators to replace them all by `/`.  I have a feeling that `pathSeparators` should be treated like more of an implementation detail, so that it is better to try to avoid relying on what it is.  Even in this case, the script is doing a blanket replace all `pathSeparators` (or, initially, the "ideal" one), trying to be agnostic to what it was supposed to be.